src/lib.rs

```diff
@@ -210,9 +210,35 @@ impl<T> From<rc::Rc<sync::Mutex<T>>> for RcMutexGuardian<T> {
 }
 
 // ****************************************************************************
-// impl ::take
+// macros
 // ****************************************************************************
 
+macro_rules! try_take {
+    ( $handle: ident, $guard:ty, $guardian:ident, $lfunc:ident ) => {{
+        use std::mem;
+        use std::sync::TryLockError::{Poisoned, WouldBlock};
+
+        // We want to express that it's safe to keep the read guard around for as long as the
+        // Arc/Rc is around. Unfortunately, we can't say this directly with lifetimes, because
+        // we have to move the Arc/Rc below, which Rust doesn't know allows the borrow to
+        // continue. We therefore transmute to a 'static Guard, and ensure that any borrows we
+        // expose are bounded by the lifetime of the guardian (which also holds the Arc/Rc).
```

Comment:
Would you mind putting this macro under the `take!` macro, and then changing this comment to say

> Safe following the same reasoning as in `take!`.